### ISSUE TITLE
accounts wiped after phone powered off

### ISSUE THREAD

**Issue by brianjmurrell** [BelledonneCommunications/linphone-android#1134]
I've had this happen more than once.  Phone powers off due to low battery and when I plug it in and turn it back on, all linphone accounts are wiped and it says I have no accounts.

Now I have to go add all of my accounts.  Again.  And re-configure each of their settings.

**VERY** annoying.  This doesn't happen with any other app on my phone after it powers down.

Here's the [log](http://brian.interlinx.bc.ca/linphone-lost-accounts.txt).  The power down happened at 21:17:55 and you can see in the log `[ series of NULLs ]` where there were a bunch of NULLs.  This would be where the power down happened.  But notice even after it powered back on, it seemed to have accounts as subsequent to the power down and turning it back on we can see:

```
2020-08-19 21:20:52:439 [org.linphone.debug/liblinphone] WARNING No proxy configuration found for local address: [sip:my_sip_account@pbx.example.com]
```

followed by many more of the same.  So how does it know about that account if it's also reporting that no accounts have been configured?

**Comment by Viish** [comment]
Can you dump the content of the following file from your device: `/data/user/0/org.linphone.debug/files/.linphonerc` ?
To do so you can use the following command: `adb shell run-as org.linphone.debug cat files/.linphonerc`

Nulls in your logs are only related to your contacts so it has no impact on your issue, and the `No proxy configuration found` is related to your chat rooms not finding the proxy config, so it's caused by your issue, not causing it.

**Comment by brianjmurrell** [comment]
```
$ adb shell run-as org.linphone.debug cat files/.linphonerc
contact="[redacted]" <sip:linphone.android@[2001:123:ab:123:f549:4f7f:9b50:a478]:51534>
use_info=0
keepalive_period=30000
sip_port=-1
sip_tcp_port=-1
sip_tls_port=-1
media_encryption=none
guess_hostname=1
register_only_when_network_is_up=1
auto_net_state_mon=1
auto_answer_replacing_calls=1
ping_with_options=0
use_cpim=1
ipv6_migration_done=1
root_ca=/data/user/0/org.linphone.debug/files/share/linphone/rootca.pem
verify_server_certs=1
verify_server_cn=1
http_proxy_port=0
default_proxy=0
zrtp_cache_migration_done=1
inc_timeout=30
in_call_timeout=0
delayed_timeout=4
register_only_when_upnp_is_ok=1
im_notif_policy=none
use_ipv6=0
use_rfc2833=1
linphone_specs=ephemeral,groupchat,lime
rls_uri=sips:rls@sip.linphone.org

[video]
size=vga
displaytype=MSAndroidTextureDisplay
display=1
capture=1
framerate=0.000000
auto_resize_preview_to_keep_ratio=1
device=FrontFacingCamera
qrcode_decoder=0
show_local=0

[app]
tunnel=disabled
push_notification=1
auto_start=1
activation_code_length=4
prefer_basic_chat_room=1
auto_download_incoming_files_max_size=-1
show_service_notification=0
push_notification_regid=[redacted]
first_launch=0
android_power_saver_dialog=1
debug=1
background_mode=1
native_dialer_call=1
link_popup_time=1557009301815
link_popup_enabled=0
device_name=FRD-L04
device_ringtone=1
random_port=1
display_contact_organization=1
store_presence_in_native_contact=0
version_check_url_last_timestamp=110838853
dark_mode=0
sender_name_hidden_in_forward_message=0
incoming_call_vibration=1
call_right_away=1
first_start=0

[tunnel]
port=443

[misc]
log_collection_upload_server_url=https://www.example.com/cgi-bin/lft.php
file_transfer_server_url=https://www.example.com/cgi-bin/lft.php
version_check_url_root=https://www.linphone.org/releases
max_calls=10
history_max_size=100
enable_basic_to_client_group_chat_room_migration=0
enable_simple_group_chat_message_state=0
aggregate_imdn=1
prefer_basic_chat_room=1
uuid=[redacted]
user_certificates_path=/data/user/0/org.linphone.debug/files/user-certs
store_friends=0
notify_each_friend_individually_when_presence_received=0
hide_chat_rooms_from_removed_proxies=0

[in-app-purchase]
server_url=https://subscribe.linphone.org:444/inapp.php
purchasable_items_ids=test_account_subscription

[net]
mtu=1300
download_bw=0
upload_bw=0
force_ice_disablement=0
nat_policy_ref=P2OjjdAoezjnfgW
friendlist_subscription_enabled=0
#Because dynamic bitrate adaption can increase bitrate, we must allow "no limit"
use_legacy_push_notification_params=1
push_notification=1
adaptive_rate_control=1

[sound]
ec_calibrator_cool_tones=1
local_ring=/data/user/0/org.linphone.debug/files/share/sounds/linphone/rings/notes_of_the_optimistic.mkv
remote_ring=/data/user/0/org.linphone.debug/files/share/sounds/linphone/ringback.wav
playback_gain_db=0.000000
mic_gain_db=0.000000
echocancellation=1
ringer_dev_id=openSLES Speaker: FRD-L04
playback_dev_id=openSLES Earpiece: FRD-L04
capture_dev_id=openSLES Microphone: FRD-L04
media_dev_id=openSLES Earpiece: FRD-L04
#remove this property for any application that is not Linphone public version itself

[assistant]
xmlrpc_url=https://subscribe.linphone.org:444/wizard.php
password_max_length=-1
password_min_length=1
username_length=-1
username_max_length=64
username_min_length=1
username_regex=^[a-z0-9+_.\-]*$
algorithm=SHA-256
domain=sip.linphone.org

[lime]
lime_update_threshold=-1
## End of factory rc
x3dh_server_url=https://lime.linphone.org/lime-server/lime-server.php

[nat_policy_0]
ref=P2OjjdAoezjnfgW

[proxy_default_values]
avpf=1
dial_escape_plus=0
publish=0
quality_reporting_collector=sip:voip-metrics@sip.linphone.org;transport=tls
quality_reporting_enabled=1
quality_reporting_interval=180
reg_expires=31536000
reg_identity=sip:?@sip.linphone.org
reg_proxy=<sip:sip.linphone.org;transport=tls>
reg_route=<sip:sip.linphone.org;transport=tls>
reg_sendregister=1
nat_policy_ref=nat_policy_default_values
realm=sip.linphone.org
conference_factory_uri=sip:conference-factory@sip.linphone.org

[nat_policy_default_values]
stun_server=stun.linphone.org
protocols=stun,ice
```

**Comment by Viish** [comment]
Indeed proxy configs are missing.
What is the result of `adb shell run-as org.linphone.debug ls -lA files/` please?

**Comment by brianjmurrell** [comment]
```
$ adb shell run-as org.linphone.debug ls -lA files/
total 12636
-rw------- 1 u0_a25 u0_a25        0 2019-03-22 15:09 .linphone.ecstate
-rw------- 1 u0_a25 u0_a25     4252 2020-08-20 08:53 .linphonerc
-rw------- 1 u0_a25 u0_a25     1855 2020-08-19 21:20 assistant_default_values
-rw------- 1 u0_a25 u0_a25     2398 2020-08-19 21:20 assistant_linphone_default_values
-rw-rw---- 1 u0_a25 u0_a25     2040 2020-05-26 09:47 default_assistant_create.rc
-rw------- 1 u0_a25 u0_a25    20480 2019-03-22 14:51 linphone-friends.db
-rw------- 1 u0_a25 u0_a25   401408 2020-08-19 14:48 linphone-log-history.db
-rw------- 1 u0_a25 u0_a25   307200 2020-07-31 12:18 linphone.db
-rw------- 1 u0_a25 u0_a25 10485807 2020-08-19 14:48 linphone1.log
-rw------- 1 u0_a25 u0_a25  1451559 2020-08-20 09:22 linphone2.log
-rw-rw---- 1 u0_a25 u0_a25     2398 2020-05-26 09:47 linphone_assistant_create.rc
-rw-rw---- 1 u0_a25 u0_a25     1119 2020-08-19 21:20 linphonerc
-rw-rw---- 1 u0_a25 u0_a25     1949 2019-03-22 14:51 lpconfig.xsd
drwx------ 5 u0_a25 u0_a25     3488 2019-03-22 14:51 share
drwx------ 2 u0_a25 u0_a25     3488 2019-03-22 14:51 user-certs
-rw------- 1 u0_a25 u0_a25   110592 2020-08-19 21:17 x3dh.c25519.sqlite3
-rw------- 1 u0_a25 u0_a25    36864 2020-08-12 16:33 zrtp_secrets
-rw------- 1 u0_a25 u0_a25    28672 2019-03-22 14:51 zrtp_secrets.bkp
```

**Comment by brianjmurrell** [comment]
Is there anything else needed here?  Any chance of resurrecting the accounts?

If no to both of the above, please let me know so that I can go re-create (again) my accounts.

**Comment by brianjmurrell** [comment]
And what can I back up to be able to easily restore the pre-wipe proxy account state in the future?

**Comment by Viish** [comment]
> Any chance of resurrecting the accounts?

No, once they are removed from the .linphonerc you can't restore them.

> Is there anything else needed here?

No, and I don't understand yet why this is happening.

> And what can I back up to be able to easily restore the pre-wipe proxy account state in the future?

The .linphonerc file contains the configuration, the linphone.db all chat messages & call history.

**Comment by brianjmurrell** [comment]
> Ok, at least it's good to know it isn't related to the app but the SDK.

How did you come to that conclusion?

**Comment by Viish** [comment]
Because @532910 said the issue happened also with latest Play Store release which is still the legacy Java app and not the kotlin rewrite. So it has to be the SDK.

**Comment by brianjmurrell** [comment]
Makes sense I guess given that the two versions of the app that have hit the issue are from completely different sources.

**Comment by Viish** [comment]
Ok I hope the proper stop of the Core when the device shutdowns will solve this issue.
Fix is in SDK 4.5.0-alpha.78+f58ab6c and newer.

**Comment by Viish** [comment]
I'm closing this issue as I expect my commit fixes it, feel free to re-open with logs if it happens again.
